graph/topo/johnson_cycles.go

```diff
@@ -107,6 +215,47 @@ func (j *johnson) circuit(v int) bool {
 	return f
 }
 
+// circuitMaxLen is the CIRCUIT sub-procedure in the paper but produces circuits up to a maximum length.
+func (j *johnson) circuitMaxLen(v int, maxLen int) bool {
+	f := false
+	// TODO: can we replace the "+ 2" with "+ 3" or "+ 4"?
+	if len(j.stack)+2 > maxLen {
+		return f
+	}
```

Comment:
If this condition is added to circuit and the order of the graph is passed in, then this method addition would not be required, true?

Reply:
yes, that is correct

Reply:
If that doesn't have a significant performance impact, I would prefer it done that way.

Reply:
It would not. I can do that